### PR TITLE
Rework Diet

### PR DESCRIPTION
**Pull request by pstalcup**
This reworks the diet to use knapsack for spleen

### CODE REVIEW

**Comment by pstalcup** [comment]
This PR needs a libram update to work correctly - it will fail builds until then

**Comment by horrible-little-slime** [comment]
> This PR needs a libram update to work correctly - it will fail builds until then

Sounds like this should be a Draft then

**Comment by pstalcup** [comment]
I am not good with computers how did this get here

File: src/diet.ts [hunk 1]
```diff
@@ -477,76 +501,14 @@ function itemPriority(menuItems: MenuItem[]) {
...
-  // Sort descending by item priority.
-  diet.sort(([x], [y]) => -(itemPriority(x) - itemPriority(y)));
-
+export function consumeDiet(diet: [MenuItem[], number][]): void {
   print();
   print("===== PLANNED DIET =====");
   printDiet(diet);
   print();
 
+  diet.sort(([x], [y]) => -(itemPriority(x) - itemPriority(y)));
```
Comment (horrible-little-slime) [inline src/diet.ts:510]:
```suggestion
  diet.sort(([a], [b]) => itemPriority(b) - itemPriority(a));
```

This is more consistent with other writing we see in garbo

File: src/diet.ts [hunk 2]
```diff
@@ -386,80 +290,200 @@ export function computeDiet(): [MenuItem[], number][] {
...
+    ...copiers(),
+
+    // FOOD POTIONS
+    ...potion($item`jumping horseradish`),
+    ...potion($item`jumping horseradish`),
+    ...potion($item`tempura cauliflower`),
+    ...potion($item`sea truffle`),
+    ...potion($item`tempura broccoli`),
+    ...limitedPotion(
+      $item`Dinsey food-cone`,
```
Comment (phulin) [inline src/diet.ts:382]: For some reason my garbo isn't getting the food-cones automatically.

Comment (phulin) [reply src/diet.ts:382]: Oh, looks like it's because it hasn't opened the zone yet.

Comment (horrible-little-slime) [reply src/diet.ts:382]: This means that we need to account for nobarf when we do that, too

File: src/potions.ts [hunk 3]
```diff
@@ -85,137 +124,217 @@ export class Potion {
...
+          quantity: limitFunction(1),
+          value: this.gross(0, ascendingTurns),
+        });
       }
     }
 
-    if (quantityToUse > 0) {
-      if (doubleDuration) quantityToUse = 1;
-      use(quantityToUse, this.potion);
+    print(`POTION: Breakpoints for ${this.potion} (starting turns ${startingTurns}):`);
```
Comment (phulin) [inline src/potions.ts:255]: Can you reduce the quantity of debug messages? Fine for debugging but I think we want less text - and more human-readable text - in the final version.

File: src/diet.ts [hunk 4]
```diff
@@ -386,80 +290,200 @@ export function computeDiet(): [MenuItem[], number][] {
...
+    return limitedPotion(potion, undefined, options);
+  }
+
+  return [
+    ...baseMenu,
+    ...copiers(),
+
+    // FOOD POTIONS
+    ...potion($item`jumping horseradish`),
+    ...potion($item`jumping horseradish`),
```
Comment (phulin) [inline src/diet.ts:377]: I don't think this should be included twice.

**Comment by phulin** [comment]
If you fix the linter errors this is good to go, right?

**Comment by phulin** [comment]
Also needs a libram update now

**Comment by pstalcup** [comment]
I'll push my latest version through tonight, we can do a libram release and it will be good to go

**Comment by phulin** [comment]
libram release is already done :)

**Comment by pstalcup** [comment]
Closing to move to a new PR to make things easier to reason about

**Pull request closed without merging**